README.template.md

```diff
@@ -25,4 +25,41 @@ Here is a quickstart you can use in your own project:
 {{ usageExampleCode }}
 ```
 
+### Error Handling
+
+Error cases in Momento are surfaced to developers as part of the return values of the method calls, as opposed
+to by throwing exceptions.  This makes them more visible, and allows your IDE to be more helpful in ensuring that
+you've handled the ones you care about.  (For more on our philosophy about this, see our blog post on why
+[Exceptions are bugs](https://www.gomomento.com/blog/exceptions-are-bugs).  And send us any feedback you have!)
+
+The preferred way of interpreting the return values from the Momento .NET methods is using [Pattern matching](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching).  Here's a quick example:
+
+```csharp
+CacheGetResponse getResponse = await client.GetAsync(CACHE_NAME, KEY);
+if (getResponse is CacheGetResponse.Hit hitResponse)
```

Comment:
What other response types are possible? Can we provide example of all possible cases or link out to external source?

Reply:
@eaddingtonwhite I added another commit that expands the treatment of this a bit further, lmk what you think.

Reply:
@eaddingtonwhite I'm going to merge this to make sure the README gets updated properly but I am more than happy to continue refining this part of the content if you have more input.

Reply:
Looks good!